uobtheatre/bookings/serializers.py

```diff
@@ -178,16 +187,78 @@ class CreateBookingSerialiser(AppendIdSerializerMixin, serializers.ModelSerializ
 
     tickets = CreateTicketSerializer(many=True, required=False)
 
+    def _check_ticket_capacities(self, tickets, performance):
+        """
+        Given the json list of tickets. Check there are enough tickets
+        available for the booking. If not return a validation error.
+        """
+
+        # Get the number of each seat group
+        seat_group_counts = {}
+        for ticket in tickets:
+            seat_group = ticket["seat_group"]
+            seat_group_count = seat_group_counts.get(seat_group)
+            seat_group_counts[seat_group] = (seat_group_count or 0) + 1
+
+        # Check each seat group is in the performance
+        seat_groups_not_in_perfromance = [
+            seat_group.name
+            for seat_group in seat_group_counts.keys()
+            if seat_group not in performance.seat_groups.all()
+        ]
+
+        # If any of the seat_groups are not assigned to this performance then throw an error
+        if len(seat_groups_not_in_perfromance) != 0:
+            return serializers.ValidationError(
+                f"You cannot book a seat group that is not assigned to this performance, you have booked {', '.join(seat_groups_not_in_perfromance)} but the performance only has {', '.join([seat_group.name for seat_group in performance.seat_groups.all()])}"
+            )
+
+        # Check that each seat group has enough capacity
+        for seat_group, number_booked in seat_group_counts.items():
+            seat_group_remaining_capacity = performance.capacity_remaining(
+                seat_group=seat_group
+            )
+            if seat_group_remaining_capacity < number_booked:
+                return serializers.ValidationError(
+                    f"There are only {seat_group_remaining_capacity} seats reamining in {seat_group} but you have booked {number_booked}. Please updated your seat selections and try again."
+                )
+
+        # Also check total capacity
+        if performance.capacity_remaining() < len(tickets):
+            return serializers.ValidationError(
+                f"There are only {performance.capacity_remaining()} seats available for this performance. You attempted to book {len(tickets)}. Please remove some tickets and try again or select a different performance."
+            )
+
+    def validate(self, attrs):
+        # If draft booking already exists
```

Comment:
Options here:
- Just delete the current draft booking if they try to create a new one
- Send an error with the existing draft booking_id/url so it can be deleted manually

Reply:
@atoff opinions?

Reply:
If a draft booking already exists, I think we should just delete old and start a new one - frontend flow:
Check for Draft Booking -> If booking, ask if user wants to resume (Y/N) -> If Y, we load and continue that booking, If N we continue as normal and send a POST to create the new booking (i.e. no DELETE for old draft)

Reply:
tldr if we post to create a new booking assume the old draft can be deleted